src/middlewared/middlewared/plugins/jail.py

```diff
@@ -706,17 +706,30 @@ def fstab(self, jail, options):
 
         verrors = ValidationErrors()
 
-        source = options.get('source')
-        if source:
-            if not os.path.exists(source):
+        if action in ('add', 'replace', 'remove'):
+            if action != 'remove' or index is None:
+                # For remove we allow removing by index or mount, so if index is not specified
+                # we should validate that rest of the fields exist.
+                for f in ('source', 'destination', 'fstype', 'fsoptions', 'dump', 'pass'):
+                    if not options.get(f):
+                        verrors.add(
+                            f'options.{f}',
+                            f'This field is required with "{action}" action.'
+                        )
+
+            if action == 'replace' and index is None:
                 verrors.add(
-                    'options.source',
-                    'Provided path for source does not exist'
+                    'options.index',
+                    'Index must not be None when replacing fstab entry.'
```

Comment:
```suggestion
                    'Index cannot be "None" when replacing an fstab entry.'
```

Reply:
If the user is ever going to see this message, it has to tell them what we mean by "index". Out of context, I don't see if index is a string or a line number, but either way, this needs to be clearer.

Reply:
@wblock @Mrt134 `index` here refers to an integer which points to the line no in fstab. Please advise how should we proceed with this. Thank you